### PR TITLE
Replace --init-config-dir with --config

### LINKED ISSUE #57763: Kubelet should take a config file, not a config dir
Jordan Liggitt and I discussed this at length last quarter. 
In short:
Handing the Kubelet a directory path and expecting users to put their kubelet configuration in a file with a magic name in that directory is less flexible, less obvious, and less user-friendly than allowing users to just specify a path to an arbitrary file. The Kubelet shouldn't care about what name a user wants on the config file, and this is more consistent with other Kubelet file-specification APIs, such as --kubeconfig.

Related: https://github.com/kubernetes/kubernetes/pull/53833#issuecomment-344009912

#57624 implements.

### PR DESCRIPTION
**Pull request by mtaufen**
Rather than a directory with magic names, just give the Kubelet a file path.
Was originally in #55718, but I'm splitting it out for clarity.

Fixes #57763

```release-note
The alpha `--init-config-dir` flag has been removed. Instead, use the `--config` flag to reference a kubelet configuration file directly.
```

### CODE REVIEW

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -325,7 +325,7 @@ func (f *KubeletFlags) AddFlags(fs *pflag.FlagSet) {
...
-	fs.Var(&f.InitConfigDir, "init-config-dir", "The Kubelet will look in this directory for the init configuration. The path may be absolute or relative; relative paths start at the Kubelet's current working directory. Omit this argument to use the built-in default configuration values. Presently, you must also enable the KubeletConfigFile feature gate to pass this flag.")
+	fs.Var(&f.KubeletConfigFile, "kubelet-config-file", "The Kubelet will load its initial configuration from this file. The path may be absolute or relative; relative paths start at the Kubelet's current working directory. Omit this flag to use the built-in default configuration values. You must also enable the KubeletConfigFile feature gate to pass this flag.")
```
Comment (liggitt) [inline cmd/kubelet/app/options/options.go]: Harmonize with --config flag in kube-proxy? kubelet prefix seems unnecessary

**Comment by liggitt** [comment]
Lgtm modulo question on flag name

File: cmd/kubelet/app/options/options.go (hunk 1 above)
Comment (mtaufen) [reply cmd/kubelet/app/options/options.go]: --pod-manifest-dir used to be --config. If that's no problem, given that the old --config is long gone, happy to harmonize. 

https://github.com/kubernetes/kubernetes/blob/release-1.4/cmd/kubelet/app/options/options.go#L87

**Comment by mtaufen** [comment]
@liggitt I decided it's probably not a big deal to use --config. Let me know what you think.

**Comment by xiangpengzhao** [comment]
/sub
This will have some influence on kubeadm. /cc @luxas

**Comment by dchen1107** [comment]
Can you summarize the reason of the decision here? Besides that, lgtm

**Comment by mtaufen** [comment]
Sure:
Handing the Kubelet a directory path and expecting users to put their kubelet configuration in a file with a magic name in that directory is less flexible, less obvious, and less user-friendly than allowing users to just specify a path to an arbitrary file. The Kubelet shouldn't care about what name a user wants on the config file, and this is more consistent with other Kubelet file-specification APIs, such as `--kubeconfig`.

**Comment by luxas** [comment]
Thanks @mtaufen, very happy with this approach :+1:
